kong/tools/batch_queue.lua

```diff
@@ -165,18 +166,19 @@ end
 -- @param self Queue
 -- @param batch: table with `entries` and `retries` counter
 -- @return nothing
-process = function(premature, self, batch)
+process = function(premature, self)
   if premature then
     return
   end
 
+  local batch = self.batch_queue[1]
```

Comment:
Do we need a sanity check here (`if not batch then ...`) in case there's nothing in the queue, or is it impossible to hit this code path with an empty queue?